bin/lib/check_reqs.js

```diff
@@ -19,15 +19,226 @@
+
+/**
+ * Gets list of installed Windows SDKs
+ *
+ * @return  {Version[]}  List of installed SDKs' versions
+ */
+function getInstalledWindowsSdks () {
+    var installedSdks = [];
+    return exec('reg query "HKLM\\SOFTWARE\\Microsoft\\Microsoft SDKs\\Windows" /s /v InstallationFolder /reg:32')
+    .fail(function () { return ''; })
+    .then(function (output) {
+        var re = /\\Microsoft SDKs\\Windows\\v(\d+\.\d+)\s*InstallationFolder\s+REG_SZ\s+(.*)/gim;
+        var match;
+        while ((match = re.exec(output))){
+            var sdkPath = match[2];
+            // Verify that SDKs is really installed by checking SDKManifest file at SDK root
+            if (shell.test('-e', path.join(sdkPath, 'SDKManifest.xml'))) {
+                installedSdks.push(Version.tryParse(match[1]));
+            }
+        }
+    })
+    .thenResolve(installedSdks);
+}
+
+/**
+ * Gets list of installed Windows Phone SDKs. Separately searches for 8.1 Phone
+ *   SDK and Windows 10 SDK, because the latter is needed for both Windows and
+ *   Windows Phone applications.
+ *
+ * @return  {Version[]}  List of installed Phone SDKs' versions.
+ */
+function getInstalledPhoneSdks () {
+    var installedSdks = [];
+    return exec('reg query "HKLM\\SOFTWARE\\Microsoft\\Microsoft SDKs\\Windows Phone\\v8.1" /v InstallationFolder /reg:32')
+    .fail(function () { return ''; })
+    .then(function (output) {
+        var match = /\\Microsoft SDKs\\Windows Phone\\v(\d+\.\d+)\s*InstallationFolder\s+REG_SZ\s+(.*)/gim.exec(output);
+        if (match && shell.test('-e', path.join(match[2], 'SDKManifest.xml'))) {
+            installedSdks.push(Version.tryParse(match[1]));
+        }
+    })
+    .then(function () {
+        return exec('reg query "HKLM\\SOFTWARE\\Microsoft\\Microsoft SDKs\\Windows\\v10.0" /v InstallationFolder /reg:32');
+    })
+    .fail(function () { return ''; })
+    .then(function (output) {
+        var match = /\\Microsoft SDKs\\Windows\\v(\d+\.\d+)\s*InstallationFolder\s+REG_SZ\s+(.*)/gim.exec(output);
+        if (match && shell.test('-e', path.join(match[2], 'SDKManifest.xml'))) {
+            installedSdks.push(Version.tryParse(match[1]));
+        }
+    })
+    .thenResolve(installedSdks);
+}
+
+/**
+ * Shortens version string or Version object by leaving only first two segments
+ *   (major and minor).
+ * @param   {String|Version}  version  The version identifier. Either Version
+ *   object or string that looks like "12.5.6"
+ * @return  {String}          Shortened version, or undefined if provided
+ *   parameter is not a valid version
+ */
+function shortenVersion (version) {
+    return /^(\d+(?:\.\d+)?)/.exec(version.toString())[1];
+}
+
+function mapWindowsVersionToName(version) {
+    var map = {
+        '6.2': 'Windows 8',
+        '6.3': 'Windows 8.1',
+        '10.0': 'Windows 10'
+    };
+    var majorMinor = shortenVersion(version);
+    return map[majorMinor];
+}
+
+function mapVSVersionToName(version) {
+    var map = {
```

Comment:
You might suggest "2012 Express for Windows", "2013 Express for Windows", "2015 Community" to be specific.  Our documentation is incorrectly specific because it says "Professional."